File: apps/macos/NativeStream/NativeStream/Views/Onboarding/OnboardingConnectionState.swift
// OnboardingConnectionState.swift

import Foundation

enum OnboardingConnectionState: Equatable {
    case idle
    case checking
    case success(channels: Int, healthy: Int, hasEpg: Bool, epgFromPlaylist: Bool)
    case failure(FailureReason)

    var isSuccess: Bool {
        if case .success = self { return true }
        return false
    }

    struct SuccessDetails: Equatable {
        let channels: Int
        let healthy: Int
        let hasEpg: Bool
        let epgFromPlaylist: Bool
    }

    var asSuccess: SuccessDetails? {
        if case .success(let channels, let healthy, let hasEpg, let epgFromPlaylist) = self {
            return SuccessDetails(
                channels: channels,
                healthy: healthy,
                hasEpg: hasEpg,
                epgFromPlaylist: epgFromPlaylist
            )
        }
        return nil
    }
}

enum FailureReason: Equatable {
    case unreachable
    case noPlaylist
    case unknown
}

File: apps/android/app/src/main/java/com/nativestream/android/ui/screens/onboarding/OnboardingConnectionState.kt
// app/src/main/java/com/nativestream/android/ui/screens/onboarding/OnboardingConnectionState.kt

package com.nativestream.android.ui.screens.onboarding

sealed class OnboardingConnectionState {
    object Idle     : OnboardingConnectionState()
    object Checking : OnboardingConnectionState()
    data class Success(
        val channels:        Int,
        val healthy:         Int,
        val hasEpg:          Boolean,
        val epgFromPlaylist: Boolean,
    ) : OnboardingConnectionState()
    data class Failure(val reason: FailureReason) : OnboardingConnectionState()
}

enum class FailureReason {
    UNREACHABLE,
    NO_PLAYLIST,
    UNKNOWN,
}